src/extractor/guidance/intersection_generator.cpp

```diff
@@ -162,12 +177,14 @@ Intersection IntersectionGenerator::GetConnectedRoads(const NodeID from_node,
     // will never happen we add an artificial invalid uturn in this case.
     if (!has_uturn_edge)
     {
-        const auto first_coordinate = coordinate_extractor.GetCoordinateAlongRoad(
-            from_node,
-            via_eid,
-            INVERT,
-            turn_node,
-            node_based_graph.GetEdgeData(via_eid).road_classification.GetNumberOfLanes());
+        const auto number_of_lanes =
+            node_based_graph.GetEdgeData(via_eid).road_classification.GetNumberOfLanes();
+        const auto first_coordinate =
+            use_low_precision_angles
+                ? coordinate_extractor.GetCoordinateCloseToTurn(
+                      from_node, via_eid, INVERT, turn_node)
+                : coordinate_extractor.GetCoordinateAlongRoad(
+                      from_node, via_eid, INVERT, turn_node, number_of_lanes);
```

Comment:
Here and above (see a, b): why not write a wrapper abstracting over selecting the function?